sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -50,6 +50,17 @@ object ParserUtils {
     }
   }
 
+  /** Check if duplicate exist in a set of column names. */
+  def checkDuplicateNames[T](colNames: Seq[String], ctx: ParserRuleContext): Unit = {
+    if (colNames.length != colNames.distinct.length) {
+      val duplicateColumns = colNames.groupBy(identity).collect {
+        case (x, ys) if ys.length > 1 => "\"" + x + "\""
+      }
+      throw new ParseException(s"Column repeated in partitioning column(s) or duplicate column " +
+        s"name key in the table definition: ${duplicateColumns.mkString("[", ",", "]")}", ctx)
```

Comment:
also, there you can give a more specific error message; the issue here is that a column is specified in both the schema and in the partitioned by clause